### PR TITLE
Add DisableActorsPreview and DisableFootprint to PlaceBuilding trait

### LINKED ISSUE #16538: Allow to disable Actor Previews in BuildingPlacement
Some mods (d2 mod) need to draw only rectangle during building placement without actual building preview.
Current code do not allow that. And overlap any PlaceBuildingDecoration trait with Actor Preview.
Need to add bool to BuildingPlacement, named DisableActorPreview with default value: False, that will prevent Actor Previews. In map editor, preview will be allowed, but in game, DisableActorPreview can be set to True.

Expected for d2 mod:
placement-expected

Current:
placement-current

( This is need for https://github.com/OpenRA/d2/issues/110 )

### PR DESCRIPTION
**Pull request by evgeniysergeev**
Closes #16538 

To test, modify /mods/d2k/rules/player.yaml and add `DisableActorPreview` to `PlaceBuilding` trait  
```
	PlaceBuilding:
		DisableActorPreview: True
```

Original:
current

Result should be like that:
after

### CODE REVIEW

File: OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs [hunk 1]
```diff
@@ -212,14 +212,21 @@ protected override IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr,
...
+							foreach (var o in api.ActorPreviewInits(actorInfo, ActorPreviewType.PlaceBuilding))
+								td.Add(o);
+
+						var init = new ActorPreviewInitializer(actorInfo, wr, td);
+						preview = actorInfo.TraitInfos<IRenderActorPreviewInfo>()
+							.SelectMany(rpi => rpi.RenderPreview(init))
+							.ToArray();
+					}
+					else
+					{
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs]: Please remove the braces around this single-line statement:
```
else
	preview = new IActorPreview[0];
```

**Comment by pchote** [comment]
IMO it would make sense to merge this into #16510 as part of a larger rewrite of that PR to introduce a `PlaceableBuilding` trait that can include flags to render the preview and footprint as well as a list of N togglable variants.

**Comment by evgeniysergeev** [comment]
Added `DisableFootprint` also, becasue it also needed for https://github.com/OpenRA/d2/issues/110

To test, modify /mods/d2k/rules/player.yaml and add DisableActorPreview to PlaceBuilding trait

```	
PlaceBuilding:
		DisableFootprint: True
```

**Comment by pchote** [comment]
I've implemented a more general approach for this in #16553. You'll be able to use `FootprintPlaceBuildingPreview` or implement your own trait if you need something more special.

**Pull request closed without merging**